conformance/lang/expressions/checking-expression/checkpanic_expr.balt

```diff
@@ -0,0 +1,1050 @@
+    io:println(a2.toBalString());
+}
+
+Test-Case: output
+Description: Test checkpanic-expr with string template expressions.
+Labels: checkpanic, string, function-defn, function-call-expr, return-stmt, optional-type, DecimalNumber, string-template-expr,
+        int
+
+function init() {
+    error? x1 = fn1(); // @output 224
+                       // @output ab
+                       // @output a
+}
+
+function fn1() returns error? {
+    string a1 = checkpanic (string`${12 + 212 - 0}`);
+    io:println(a1);
+    string a2 = checkpanic (string`${"a" + "b"}`);
+    io:println(a2);
+    string a3 = checkpanic (string`${string `${"a"}`}`);
+    io:println(a3);
+}
+
+Test-Case: error
+Description: Test the static type of the checkpanic-expr when it is used with string template expressions.
+Labels: decimal, DecimalNumber, float, string, function-defn, function-call-expr, return-stmt, optional-type, checkpanic,
+        string-template-expr, error-type, int
+
+function errorFunction() {
+    error? x1 = fn1(); // @error expected '(int|error)' but found 'string'
+                       // @error expected '(float|error)' but found 'string'
+                       // @error expected '(decimal|error)' but found 'string'
+}
+
+function fn1() returns error? {
+    int _ = checkpanic (string`${12 + 212 - 0}`);
+    float _ = checkpanic (string`${"a" + "b"}`);
+    decimal _ = checkpanic (string`${string `${"a"}`}`);
+}
+
+Test-Case: output
+Description: Test checkpanic-expr with structural constructor expressions.
+Fail-Issue: ballerina-platform/ballerina-lang#35162
+Labels: array-type, decimal, DecimalFloatingPointNumber, DecimalNumber, float, FloatingPointTypeSuffix, HexIntLiteral,
+        list-constructor-expr, mapping-constructor-expr, string, byte, int:MAX_VALUE, int:MIN_VALUE, function-defn,
+        function-call-expr, return-stmt, union-type, optional-type, map-type, readonly-type, checkpanic, spread-field, unary-minus,
+        table-constructor-expr, table-type, record-type, checkpanic, int
+
+function init() {
+    error? x1 = listConstructor(); // @output [1,2,3,-18,-9223372036854775808]
+                                   // @output [1.2,-0.21,102,-21]
+                                   // @output ["1","b","+"]
+                                   // @output [["1","b","+"]]
+                                   // @output [["1","b","+"]]
+
+    error? x2 = mappingConstructor(); // @output {"a":1,"b":12,"3":5,"max":9223372036854775807}
+                                   // @output {"x":1.21,"y":2.0,"z":-212}
+                                   // @output {"0":0,"255":255,"hex":18}
+                                   // @output {"b":"b","a":"a"}
+                                   // @output {"b":"b","a":"a"}
+
+    error? x3 = tableConstructor(); // @output [{"name":"John","salary":100},{"name":"Jane","salary":200.2}]
+}
+
+function listConstructor() returns error? {
+    int[] a1 = checkpanic [1, 2, 3, -0x12, int:MIN_VALUE];
```

Comment:
Shall we add test cases like,

`int[] a1 = checkpanic [1, 2, 0/0, -0x12, int:MIN_VALUE];`

`int[] a1 = [1, 2, checkpanic 0/0, -0x12, int:MIN_VALUE];`

Reply:
I think, `checkpanic` is not necessary because `0/0` causes to panic the code. will add this with trap.